ui/tests/acceptance/optimize-test.js

```diff
@@ -351,6 +351,339 @@ module('Acceptance | optimize', function(hooks) {
+    paramName: 'status',
+    expectedOptions: ['Pending', 'Running', 'Dead'],
+    async beforeEach() {
+      server.createList('job', 2, {
+        status: 'pending',
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.createList('job', 2, {
+        status: 'running',
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.createList('job', 2, { status: 'dead', createRecommendations: true, childrenCount: 0 });
+      await Optimize.visit();
+    },
+    filter: (taskGroup, selection) => selection.includes(taskGroup.job.status),
+  });
+
+  testFacet('Datacenter', {
+    facet: Optimize.facets.datacenter,
+    paramName: 'dc',
+    expectedOptions(jobs) {
+      const allDatacenters = new Set(
+        jobs.mapBy('datacenters').reduce((acc, val) => acc.concat(val), [])
+      );
+      return Array.from(allDatacenters).sort();
+    },
+    async beforeEach() {
+      server.create('job', {
+        datacenters: ['pdx', 'lax'],
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.create('job', {
+        datacenters: ['pdx', 'ord'],
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.create('job', {
+        datacenters: ['lax', 'jfk'],
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.create('job', {
+        datacenters: ['jfk', 'dfw'],
+        createRecommendations: true,
+        groupsCount: 1,
+        groupTaskCount: 2,
+        childrenCount: 0,
+      });
+      server.create('job', { datacenters: ['pdx'], createRecommendations: true, childrenCount: 0 });
+      await Optimize.visit();
+    },
+    filter: (taskGroup, selection) => taskGroup.job.datacenters.find(dc => selection.includes(dc)),
+  });
+
+  testFacet('Prefix', {
+    facet: Optimize.facets.prefix,
+    paramName: 'prefix',
+    expectedOptions: ['hashi (3)', 'nmd (2)', 'pre (2)'],
+    async beforeEach() {
+      [
+        'pre-one',
+        'hashi_one',
+        'nmd.one',
+        'one-alone',
+        'pre_two',
+        'hashi.two',
+        'hashi-three',
+        'nmd_two',
+        'noprefix',
+      ].forEach(name => {
+        server.create('job', {
+          name,
+          createRecommendations: true,
+          createAllocations: true,
+          groupsCount: 1,
+          groupTaskCount: 2,
+          childrenCount: 0,
+        });
+      });
+      await Optimize.visit();
+    },
+    filter: (taskGroup, selection) => selection.find(prefix => taskGroup.job.name.startsWith(prefix)),
+  });
+
+  function testFacet(label, { facet, paramName, beforeEach, filter, expectedOptions }) {
```

Comment:
This is another one of those things where there just wasn't quite a good way to abstract it when I introduced the pattern for clients and jobs. Would be nice to simplify somehow.